server/src/db.ts

```diff
@@ -70,3 +91,57 @@ export async function editZone(zone: Zone): Promise<void> {
 export async function deleteZone(id: number): Promise<void> {
   await dbRun('DELETE FROM zones WHERE id = ?', [id]);
 }
+
+// ---- Light Operations ----
+
+export async function getLights(): Promise<Light[]> {
+  return dbAll(`SELECT * FROM lights`) as Promise<Light[]>;
+}
+
+export async function createLight(
+  lightRequest: CreateLightRequest
+): Promise<void> {
+  switch (lightRequest.type) {
+    case LightType.RVL: {
+      const rvlLightRequest: CreateRVLLightRequest = lightRequest as CreateRVLLightRequest;
+      await dbRun(`INSERT INTO lights (name, type, channel) values (?, ?, ?)`, [
+        rvlLightRequest.name,
+        LightType.RVL,
+        rvlLightRequest.channel
+      ]);
+      break;
+    }
+    case LightType.Hue: {
+      const hueLightRequest: CreateHueLightRequest = lightRequest as CreateHueLightRequest;
+      await dbRun(`INSERT INTO lights (name, type) values (?, ?)`, [
```

Comment:
Remind me, does this lib protect from sql injection if you parameterize like this?

Reply:
Yes it does, see https://github.com/mapbox/node-sqlite3/wiki/API#databaserunsql-param--callback